src/clj/puppetlabs/services/request_handler/request_handler_service.clj

```diff
@@ -3,18 +3,28 @@
             [puppetlabs.services.protocols.request-handler :as handler]
             [puppetlabs.services.request-handler.request-handler-core :as core]
             [puppetlabs.services.jruby.jruby-puppet-service :as jruby]
-            [puppetlabs.trapperkeeper.services :as tk-services]))
+            [puppetlabs.trapperkeeper.services :as tk-services]
+            [clojure.tools.logging :as log]))
 
 (defn- handle-request
-  [request jruby-service config]
-  (jruby/with-jruby-puppet jruby-puppet jruby-service
-    (core/handle-request request jruby-puppet config)))
+  [request jruby-service config puppet-version]
+  (try
+    (jruby/with-jruby-puppet jruby-puppet jruby-service
+                             (core/handle-request request jruby-puppet config))
+    (catch IllegalStateException e
```

Comment:
Offhand, I don't have a preference where the error handling lives.

However, I think it's wrong translate all `IllegalStateExceptions` into HTTP 503s.  In general, I would expect an `IllegalStateException` to be an HTTP 500.  Since the one particular error we actually want to catch here is thrown in our code (when the `borrow` times-out), I think it should throw a slingshot error, which can be caught instead and translated into an HTTP 503.